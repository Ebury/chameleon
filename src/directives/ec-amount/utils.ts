
import defaultOptions from './options';
import type { AmountDirectiveOptions } from './types';

function format(input: undefined, opts?: AmountDirectiveOptions): undefined;
function format(input: null, opts?: AmountDirectiveOptions): null;
function format(input: string | number, opts?: AmountDirectiveOptions): string;
function format(input: string | number | undefined | null, opts: AmountDirectiveOptions = defaultOptions): string | null | undefined {
  /* eslint-disable no-param-reassign */
  if (input === '' || input === null || input === undefined) {
    return input;
  }

  if (typeof input === 'number') {
    input = input.toString();
  }

  const negative = input.startsWith('-') ? '-' : '';

  input = sanitizeInput(input, opts.decimalSeparator);
  if (opts.precision > 0) {
    input = keepOnlyOneDecimalSeparator(input, opts.decimalSeparator);
  } else {
    input = removeDecimal(input, opts.decimalSeparator);
  }
  const parts = input.split(opts.decimalSeparator);
  let integralPart = parseToStr(parts[0]);
  let fractionalPart = parts[1];

  if (fractionalPart && fractionalPart.length > opts.precision) {
    fractionalPart = fractionalPart.substring(0, opts.precision);
  }

  integralPart = addGroupingSeparators(integralPart, opts.groupingSeparator);
  const result = negative + joinIntegralAndFractionalParts(integralPart, fractionalPart, opts.decimalSeparator);
  return result;
}

function parseToStr(number: string): string {
  const parsed = Number.parseFloat(number);
  if (Number.isNaN(parsed)) {
    return '';
  }
  return parsed.toString();
}

function keepOnlyOneDecimalSeparator(number: string, decimalSeparator: string): string {
  if (number.includes(decimalSeparator)) {
    const split = number.split(decimalSeparator);
    return split[0] + decimalSeparator + split.slice(1).join('');
  }
  return number;
}

function removeDecimal(number: string, decimalSeparator: string): string {
  const split = number.split(decimalSeparator);
  return split[0];
}

function unFormat(input: string, groupingSeparator: string, decimalSeparator: string): string {
  if (!input) {
    return input;
  }

  return input
    .replace(new RegExp(`\\${groupingSeparator}`, 'g'), '')
    .replace(new RegExp(`\\${decimalSeparator}`, 'g'), '.');
}

function sanitizeInput(input: string, decimalSeparator: string): string {
  return input.replace(new RegExp(`[^0-9${decimalSeparator}]`, 'g'), '') || '';
}

function addGroupingSeparators(integralPart: string, separator: string): string {
  return integralPart.replace(/(\d)(?=(?:\d{3})+\b)/gm, `$1${separator}`);
}

function joinIntegralAndFractionalParts(integralPart: string, fractionalPart: string, decimalSeparator: string): string {
  return fractionalPart !== undefined ? integralPart + decimalSeparator + fractionalPart : integralPart;
}

/* c8 ignore start */
function setCursor(el: HTMLInputElement, position: number) {
  const setSelectionRange = function setSelectionRange() {
    el.setSelectionRange(position, position);
  };

  if (el === document.activeElement) {
    setSelectionRange();
    setTimeout(setSelectionRange, 1); // Android Fix
  }
}
/* c8 ignore stop */

export {
  format,
  sanitizeInput,
  setCursor,
  unFormat,
};
