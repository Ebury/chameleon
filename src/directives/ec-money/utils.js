import defaults from './options';

function format(input, opt = defaults) {
  if (input === '' || input === null || input === undefined) {
    return input;
  }

  if (typeof input === 'number') {
    input = input.toString();
  }

  const negative = input.includes('-') ? '-' : '';

  input = sanitizeInput(input, opt.decimalSeparator);
  input = keepOnlyOneDecimalSeparator(input, opt.decimalSeparator);
  const parts = input.split(opt.decimalSeparator);
  let integer = parseToStr(parts[0]);
  let decimal = parts[1];

  if (decimal && decimal.length > opt.precision) {
    decimal = decimal.substr(0, opt.precision);
  }

  integer = addGroupingSeparators(integer, opt.groupingSeparator);
  const result = negative + joinIntegerAndDecimal(integer, decimal, opt.decimalSeparator);
  return result;
}

function parseToStr(number) {
  /* eslint no-restricted-globals: "off" */
  const parsed = parseFloat(number);
  if (isNaN(parsed)) {
    return '';
  }
  return parsed.toString();
}

function keepOnlyOneDecimalSeparator(number, decimalSeparator) {
  if (number.includes(decimalSeparator)) {
    const split = number.split(decimalSeparator);
    return split[0] + decimalSeparator + split.slice(1).join('');
  }
  return number;
}

function unFormat(input, groupingSeparator, decimalSeparator) {
  /* eslint no-param-reassign: "off" */
  input = (`${input}`).replace(new RegExp(`\\${groupingSeparator}`, 'g'), '');
  input = (`${input}`).replace(new RegExp(`\\${decimalSeparator}`, 'g'), '.');
  return input;
}

function sanitizeInput(input, decimalSeparator) {
  return input.replace(new RegExp(`[^0-9${decimalSeparator}]`, 'g'), '') || '';
}

function addGroupingSeparators(integer, separator) {
  return integer.replace(/(\d)(?=(?:\d{3})+\b)/gm, `$1${separator}`);
}

function joinIntegerAndDecimal(integer, decimal, decimalSeparator) {
  return decimal !== undefined ? integer + decimalSeparator + decimal : integer;
}

function setCursor(el, position) {
  const setSelectionRange = function setSelectionRange() {
    el.setSelectionRange(position, position);
  };

  if (el === document.activeElement) {
    setSelectionRange();
    setTimeout(setSelectionRange, 1); // Android Fix
  }
}

// https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Creating_and_triggering_events#The_old-fashioned_way
function event(name) {
  const evt = document.createEvent('Event');
  evt.initEvent(name, true, true);
  return evt;
}

export {
  format,
  unFormat,
  setCursor,
  event,
  sanitizeInput,
};
