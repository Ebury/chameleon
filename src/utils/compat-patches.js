/* istanbul ignore file */

import {
  Dropdown as FvDropdown,
  TooltipDirective as FvTooltipDirective,
  VClosePopper,
  VTooltip,
} from 'floating-vue';

// we need to tell the @vue/compat that all components coming from floating-vue are Vue 3 components so it skips
// them when patching component instances. These components can't run in default mode for Vue 2, there is lots of
// errors coming from the console in that mode.
export default function applyCompatPatches() {
  const compatConfig = {
    MODE: 3,
    RENDER_FUNCTION: false,
  };

  VTooltip.compatConfig = compatConfig;
  VClosePopper.compatConfig = compatConfig;
  FvTooltipDirective.compatConfig = compatConfig;
  for (const component of Object.values(FvTooltipDirective.components)) {
    component.compatConfig = compatConfig;
  }

  FvDropdown.compatConfig = compatConfig;

  for (const component of Object.values(FvDropdown.components)) {
    component.compatConfig = compatConfig;
  }
}
