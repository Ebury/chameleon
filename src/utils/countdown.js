import mitt from 'mitt';

export default class Countdown {
  constructor() {
    const m = mitt();
    Object.assign(this, m);
  }

  reduceMinutesLeft(totalSeconds) {
    this.getminutes(totalSeconds);
    this.getseconds(totalSeconds, this.minutesLeft);
  }

  start(secondsToGo, showMinutes) {
    this.seconds = secondsToGo;
    this.secondsLeft = secondsToGo;
    this.currentTime = Math.ceil(Date.now() / 1000);
    this.startTime = Math.ceil(Date.now() / 1000);
    this.interval = setInterval(() => this.reduceSecondsLeft(showMinutes), 1000);
  }

  get timeDifference() {
    return this.currentTime - this.startTime;
  }

  reduceSecondsLeft(showMinutes) {
    this.currentTime = Math.ceil(Date.now() / 1000);
    this.totalSeconds = this.seconds - this.timeDifference;
    this.minutesLeft = null;
    this.secondsLeft = this.totalSeconds;

    if (showMinutes) {
      this.reduceMinutesLeft(this.totalSeconds);
    }

    this.emit('time-updated', {
      seconds: this.secondsLeft,
      minutes: this.minutesLeft,
      totalSeconds: this.totalSeconds,
    });

    if (this.totalSeconds <= 0) {
      this.stop();
      this.emit('time-expired');
    }
  }

  stop() {
    clearInterval(this.interval);
  }

  getminutes(totalSeconds) {
    this.minutesLeft = Math.floor(totalSeconds / 60);
  }

  getseconds(totalSeconds, minutes) {
    // take mins remaining (as seconds) away from total seconds remaining
    this.secondsLeft = totalSeconds - Math.round(minutes * 60);
  }
}
