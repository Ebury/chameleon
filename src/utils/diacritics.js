export function removeDiacritics(str) {
  if (typeof str === 'string' && str.normalize) {
    // https://stackoverflow.com/questions/990904/remove-accents-diacritics-in-a-string-in-javascript
    return str.normalize('NFD').replace(/[\u0300-\u036f]/g, '');
  }

  // IE11 users will not enjoy this feature:
  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize
  return str;
}
