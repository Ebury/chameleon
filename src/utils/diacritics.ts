export function removeDiacritics(str: string): string {
  if (str && str.normalize) {
    // https://stackoverflow.com/questions/990904/remove-accents-diacritics-in-a-string-in-javascript
    return str.normalize('NFD').replace(/[\u0300-\u036f]/g, '');
  }

  // IE11 users will not enjoy this feature:
  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize
  return str;
}
