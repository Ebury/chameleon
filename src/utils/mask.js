export function mask(text, maskSymbol = '*', visibleChars = 2) {
  if (!text) {
    throw new Error('Text is required');
  }
  if (!maskSymbol) {
    throw new Error('Mask symbol cannot be empty');
  }
  if (!visibleChars || visibleChars < 0) {
    throw new Error('Visible characters must be a number greater than zero');
  }

  const digitsToBeMasked = text.length - visibleChars;
  return maskSymbol.repeat(digitsToBeMasked) + text.substr(digitsToBeMasked);
}
