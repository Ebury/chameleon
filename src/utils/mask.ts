export function mask(text: string, maskSymbol: string = '*', visibleChars: number = 2): string {
  if (!text) {
    throw new Error('Text is required');
  }
  if (!maskSymbol) {
    throw new Error('Mask symbol cannot be empty');
  }
  if (!visibleChars || visibleChars < 0) {
    throw new Error('Visible characters must be a number greater than zero');
  }

  const digitsToBeMasked = text.length - visibleChars;
  return maskSymbol.repeat(digitsToBeMasked) + text.substring(digitsToBeMasked);
}
