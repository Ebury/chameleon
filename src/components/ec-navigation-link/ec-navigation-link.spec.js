import { mount } from '@vue/test-utils';
import EcNavigationLink from './ec-navigation-link.vue';
import { withMockedConsole } from '../../../tests/utils/console';

describe('EcNavigationLink', () => {
  it('should throw an error if prop text was not given', () => {
    withMockedConsole((errorSpy) => {
      mount(EcNavigationLink);
      expect(errorSpy).toHaveBeenCalled();
      expect(errorSpy.mock.calls[0][0]).toContain('Missing required prop: "text"');
    });
  });

  it('should throw an error if prop url was not given', () => {
    withMockedConsole((errorSpy) => {
      mount(EcNavigationLink, {
        propsData: {
          text: 'Random text',
        },
      });
      expect(errorSpy).toHaveBeenCalled();
      expect(errorSpy.mock.calls[0][0]).toContain('Missing required prop: "url"');
    });
  });

  describe('as router-link', () => {
    function mountAsRouterLink(opts, mountOpts) {
      const propsData = {
        text: 'Link',
        iconName: 'simple-check',
        url: '/balances',
        isRouterLink: true,
        ...opts,
      };

      return mount(EcNavigationLink, {
        propsData,
        ...mountOpts,
      });
    }

    it('should render as a router link if isRouterLink prop set to true', () => {
      const wrapper = mountAsRouterLink({ isRouterLink: true });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should be expanded by default', () => {
      const wrapper = mountAsRouterLink();
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.classes('ec-navigation-link--is-collapsed')).toBe(false);
    });

    it('should hide the text when is collapsed', () => {
      const wrapper = mountAsRouterLink({ isCollapsed: true });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.findByDataTest('ec-navigation-link__text').isVisible()).toBe(false);
    });

    it('should show the text when is expanded', () => {
      const wrapper = mountAsRouterLink({ isCollapsed: false });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.findByDataTest('ec-navigation-link__text').isVisible()).toBe(true);
    });

    it('should be active when isActive is passed into', () => {
      const wrapper = mountAsRouterLink({ isActive: true });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.classes('ec-navigation-link--is-active')).toBe(true);
    });

    it('should be compact when isCompact is passed into', () => {
      const wrapper = mountAsRouterLink({ isCompact: true });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.classes('ec-navigation-link--is-compact')).toBe(true);
    });
  });

  describe('as regular anchor', () => {
    function mountAsAnchor(opts, mountOpts) {
      const propsData = {
        text: 'Link',
        iconName: 'simple-check',
        url: '/balances',
        isRouterLink: false,
        ...opts,
      };

      return mount(EcNavigationLink, {
        propsData,
        ...mountOpts,
      });
    }

    it('should render as a normal <a> tag if isRouterLink is set to false', () => {
      const wrapper = mountAsAnchor({ isRouterLink: false });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should be expanded by default', () => {
      const wrapper = mountAsAnchor();
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.classes('ec-navigation-link--is-collapsed')).toBe(false);
    });

    it('should hide the text when is collapsed', () => {
      const wrapper = mountAsAnchor({ isCollapsed: true });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.findByDataTest('ec-navigation-link__text').isVisible()).toBe(false);
    });

    it('should show the text when is expanded', () => {
      const wrapper = mountAsAnchor({ isCollapsed: false });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.findByDataTest('ec-navigation-link__text').isVisible()).toBe(true);
    });

    it('should be active when isActive is passed into', () => {
      const wrapper = mountAsAnchor({ isActive: true });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.classes('ec-navigation-link--is-active')).toBe(true);
    });

    it('should be compact when isCompact is passed into', () => {
      const wrapper = mountAsAnchor({ isCompact: true });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.classes('ec-navigation-link--is-compact')).toBe(true);
    });

    it('should pass listeners from parent to the root', () => {
      const clickSpy = jest.fn();
      const wrapper = mountAsAnchor({}, {
        listeners: { click: clickSpy },
      });

      wrapper.trigger('click');
      expect(clickSpy).toHaveBeenCalledTimes(1);
    });
  });
});
