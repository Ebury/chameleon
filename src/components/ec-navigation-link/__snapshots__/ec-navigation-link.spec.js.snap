// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcNavigationLink as regular anchor should be active when isActive is passed into 1`] = `
<a
  class="ec-navigation-link ec-navigation-link--is-active"
  data-test="ec-navigation-link"
  href="/balances"
>
  <svg
    class="ec-navigation-link__icon ec-icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-single-check"
    />
  </svg>
   
  <span
    class="ec-navigation-link__text"
    data-test="ec-navigation-link__text"
  >
    Link
  </span>
</a>
`;

exports[`EcNavigationLink as regular anchor should be compact when isCompact is passed into 1`] = `
<a
  class="ec-navigation-link ec-navigation-link--is-compact"
  data-test="ec-navigation-link"
  href="/balances"
>
  <svg
    class="ec-navigation-link__icon ec-icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-single-check"
    />
  </svg>
   
  <span
    class="ec-navigation-link__text"
    data-test="ec-navigation-link__text"
  >
    Link
  </span>
</a>
`;

exports[`EcNavigationLink as regular anchor should be expanded by default 1`] = `
<a
  class="ec-navigation-link"
  data-test="ec-navigation-link"
  href="/balances"
>
  <svg
    class="ec-navigation-link__icon ec-icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-single-check"
    />
  </svg>
   
  <span
    class="ec-navigation-link__text"
    data-test="ec-navigation-link__text"
  >
    Link
  </span>
</a>
`;

exports[`EcNavigationLink as regular anchor should hide the text when is collapsed 1`] = `
<a
  class="ec-navigation-link ec-navigation-link--is-collapsed"
  data-test="ec-navigation-link"
  href="/balances"
>
  <svg
    class="ec-navigation-link__icon ec-icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-single-check"
    />
  </svg>
   
  <span
    class="ec-navigation-link__text"
    data-test="ec-navigation-link__text"
    style="display: none;"
  >
    Link
  </span>
</a>
`;

exports[`EcNavigationLink as regular anchor should render as a normal <a> tag if isRouterLink is set to false 1`] = `
<a
  class="ec-navigation-link"
  data-test="ec-navigation-link"
  href="/balances"
>
  <svg
    class="ec-navigation-link__icon ec-icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-single-check"
    />
  </svg>
   
  <span
    class="ec-navigation-link__text"
    data-test="ec-navigation-link__text"
  >
    Link
  </span>
</a>
`;

exports[`EcNavigationLink as regular anchor should show the text when is expanded 1`] = `
<a
  class="ec-navigation-link"
  data-test="ec-navigation-link"
  href="/balances"
>
  <svg
    class="ec-navigation-link__icon ec-icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-single-check"
    />
  </svg>
   
  <span
    class="ec-navigation-link__text"
    data-test="ec-navigation-link__text"
  >
    Link
  </span>
</a>
`;

exports[`EcNavigationLink as router-link should be active when isActive is passed into 1`] = `
<router-link-stub
  active-class="ec-navigation-link--is-active"
  class="ec-navigation-link ec-navigation-link--is-active"
  data-test="ec-navigation-link"
  to="/balances"
>
  <svg
    class="ec-navigation-link__icon ec-icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-single-check"
    />
  </svg>
   
  <span
    class="ec-navigation-link__text"
    data-test="ec-navigation-link__text"
  >
    Link
  </span>
</router-link-stub>
`;

exports[`EcNavigationLink as router-link should be compact when isCompact is passed into 1`] = `
<router-link-stub
  active-class="ec-navigation-link--is-active"
  class="ec-navigation-link ec-navigation-link--is-compact"
  data-test="ec-navigation-link"
  to="/balances"
>
  <svg
    class="ec-navigation-link__icon ec-icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-single-check"
    />
  </svg>
   
  <span
    class="ec-navigation-link__text"
    data-test="ec-navigation-link__text"
  >
    Link
  </span>
</router-link-stub>
`;

exports[`EcNavigationLink as router-link should be expanded by default 1`] = `
<router-link-stub
  active-class="ec-navigation-link--is-active"
  class="ec-navigation-link"
  data-test="ec-navigation-link"
  to="/balances"
>
  <svg
    class="ec-navigation-link__icon ec-icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-single-check"
    />
  </svg>
   
  <span
    class="ec-navigation-link__text"
    data-test="ec-navigation-link__text"
  >
    Link
  </span>
</router-link-stub>
`;

exports[`EcNavigationLink as router-link should hide the text when is collapsed 1`] = `
<router-link-stub
  active-class="ec-navigation-link--is-active"
  class="ec-navigation-link ec-navigation-link--is-collapsed"
  data-test="ec-navigation-link"
  to="/balances"
>
  <svg
    class="ec-navigation-link__icon ec-icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-single-check"
    />
  </svg>
   
  <span
    class="ec-navigation-link__text"
    data-test="ec-navigation-link__text"
    style="display: none;"
  >
    Link
  </span>
</router-link-stub>
`;

exports[`EcNavigationLink as router-link should render as a router link if isRouterLink prop set to true 1`] = `
<router-link-stub
  active-class="ec-navigation-link--is-active"
  class="ec-navigation-link"
  data-test="ec-navigation-link"
  to="/balances"
>
  <svg
    class="ec-navigation-link__icon ec-icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-single-check"
    />
  </svg>
   
  <span
    class="ec-navigation-link__text"
    data-test="ec-navigation-link__text"
  >
    Link
  </span>
</router-link-stub>
`;

exports[`EcNavigationLink as router-link should show the text when is expanded 1`] = `
<router-link-stub
  active-class="ec-navigation-link--is-active"
  class="ec-navigation-link"
  data-test="ec-navigation-link"
  to="/balances"
>
  <svg
    class="ec-navigation-link__icon ec-icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-single-check"
    />
  </svg>
   
  <span
    class="ec-navigation-link__text"
    data-test="ec-navigation-link__text"
  >
    Link
  </span>
</router-link-stub>
`;
